nixos/modules/security/apparmor.nix

```
@@ -42,22 +51,55 @@ in
         description = ''
           AppArmor policies.
         '';
-        type = types.attrsOf (types.submodule ({ name, config, ... }: {
-          options = {
-            enable = mkDisableOption "loading of the profile into the kernel";
-            enforce = mkDisableOption "enforcing of the policy or only complain in the logs";
-            profile = lib.mkOption {
-              description = "The policy of the profile.";
-              type = types.lines;
-              apply = pkgs.writeText name;
-            };
-          };
-        }));
-        default = {};
+        type = types.attrsOf (
+          types.submodule (
+            {
+              name,
+              config,
+              options,
+              ...
+            }:
+            {
+              options = {
+                state = lib.mkOption {
+                  description = "How strictly this policy should be enforced";
+                  type = types.enum [
+                    "disable"
+                    "complain"
+                    "enforce"
+                  ];
+                  # should enforce really be the default?
+                  # the docs state that this should only be used once one is REALLY sure nothing's gonna break
+                  default = "enforce";
+                };
+
+                profile = lib.mkOption {
+                  description = "The profile file contents. Incompatible with path.";
+                  type = types.lines;
+                };
+
+                path = lib.mkOption {
+                  type = types.nullOr types.path;
+                  default = null;
+                  description = "A path of a profile file to include. Incompatible with profile.";
```

```suggestion
                  description = "A path of a profile file to include. Incompatible with profile.";
                  type = types.nullOr types.path;
                  default = null;
```
Nitpick for consistency with other declarations